Assets/Scripts/Systems/Modules/ModulesInfo.cs

```diff
@@ -0,0 +1,175 @@
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+using Ships.Components;
+using Systems.Save;
+using UnityEngine;
+
+namespace Systems.Modules
+{
+    /// <summary>
+    ///     Stores runtime information on Modules and Module Grid.
+    ///     Saves and Loads Modules and Module Grid.
+    ///     Will init with default values from ShipData if there is no save data.
+    /// </summary>
+    [Serializable]
+    public class ModulesInfo : MonoBehaviour, ISavable
+    {
+        private Module[,] _grid;
+        private ModuleSlot[,] _slotGrid;
+        private List<Module> _modules;
+        private List<ModuleSlot> _moduleSlots;
+        private int _rowHeight;
+        private int _columnLength;
+
+        private ShipInfo _info;
+
+        public string id => "module_grid";
+        public int RowHeight => _rowHeight;
+        public int ColumnLength => _columnLength;
+        public List<Module> Modules => _modules;
+
+        private void Awake()
+        {
+            _info = GetComponent<ShipInfo>();
+            if (_grid == null)
+            {
+                _rowHeight = _info.Data.ModuleGridHeight;
+                _columnLength = _info.Data.ModuleGridWidth;
+                _grid = new Module[_rowHeight, _columnLength];
+                _modules = _info.Data.ModuleList;
+                _moduleSlots = _info.Data.ModuleSlots;
+                foreach (Module module in _modules)
+                {
+                    AddModule(module);
+                }
+
+                foreach (ModuleSlot moduleSlot in _moduleSlots)
+                {
+                    AddModuleSlot(moduleSlot);
+                }
+            }
+        }
+
+        public Module GetModule(Coordinates modulePos)
+        {
+            return _grid[modulePos.y, modulePos.x];
+        }
+
+        public bool AddModule(Module module)
```

Comment:
I feel like this should take ModuleData and Coordinates, since this is generally how you are adding modules.